books_iterator/main.py

```diff
@@ -32,48 +37,77 @@ class Price(object):
         },
     }
 
+
     def __init__(self, amount, currency='USD'):
-        pass
+        self.amount=amount
+        self.currency=currency
 
     def __str__(self):
-        pass
+        return "Price(Decimal('{}'), '{}')".format(self.amount,self.currency)
 
     def get_currency(self):
-        pass
+        return self.currency
 
     def __add__(self, other):
         # return a new `Price` instance, representing the sum of
         # both given ones
-        pass
+        if self.currency==other.currency:
+            sum_amount=self.amount+other.amount
+        else: 
+            sum_amount=self.amount+other.get_value(self.currency)
+        return Price(sum_amount,self.currency)
 
     def __eq__(self, other):
         # compare if two prices are equal. Keep in mind that both prices
         # might have different currencies. Use the `.get_value()` function
         # to transform prices to a comparable currency.
-        pass
+        if self.currency==other.currency and self.amount==other.amount:
+            return True
+        elif self.currency!=other.currency:
+            if self.amount==other.get_value(self.currency):
+                return True
 
     def __ne__(self, other):
         # opposite to __eq__
-        pass
+        if self.currency==other.currency and self.amount==other.amount:
+            return True
+        elif self.currency!=other.currency:
+            if self.amount==other.get_value(self.currency):
+                return True
 
     def get_value(self, currency=None):
         # if no currency is given, returns the current price amount. If a
         # different currency is given, handles the price convertion to the
         # given currency. Use the `EXCHANGE_RATES` dict for that.
-        pass
+        if not currency:
+            return self.amount
+        if currency=='USD':
+            return Decimal(self.amount*(self.EXCHANGE_RATES[self.currency]['USD']))
+        elif currency=='EUR':
+            return Decimal(self.amount*(self.EXCHANGE_RATES[self.currency]['EUR']))
+        elif currency=='YEN':
+            return Decimal(self.amount*(self.EXCHANGE_RATES[self.currency]['YEN']))
 
 
 class BookIterator(object):
     def __init__(self, file_path):
-        pass
+        self.file_path=file_path
+        self.items=[Book(x[0],x[1],x[2],x[3]) for x in read_file_line_by_line(self.file_path)]
```

Comment:
the method `read_file_line_by_line(self.file_path)` itself is already returning a list of objects, you don't need to do it again. 

`self.items = read_file_line_by_line(self.file_path)`